syft/frameworks/torch/mpc/fss.py

```diff
@@ -41,7 +41,58 @@ def full_name(f):
 
 # number of processes
 N_CORES = 8
-MULTI_LIMIT = 10_000
+MULTI_LIMIT = 50_000
+
+
+def keygen(n_values, op):
+    """
+    Run FSS keygen in parallel to accelerate the offline part of the protocol
+
+    Args:
+        n_values (int): number of primitives to generate
+        op (str): eq or comp <=> DPF or DIF
+    """
+    if op == "eq":
+        return DPF.keygen(n_values=n_values)
+    elif op == "comp":
+        if n_values > MULTI_LIMIT:
+            # print('MULTI EVAL', numel, x_masked.owner)
+            multiprocessing_args = []
+            slice_size = math.ceil(n_values / N_CORES)
+            for j in range(N_CORES):
+                n_instances = min((j + 1) * slice_size, n_values) - j * slice_size
+                process_args = (n_instances,)  # TODO add a seed element for the PRG?
+                multiprocessing_args.append(process_args)
+            p = multiprocessing.Pool()
+            partitions = p.starmap(DIF.keygen, multiprocessing_args)
+            p.close()
```

Comment:
Q: It would be better to call ```join```?
Seen [this](https://stackoverflow.com/questions/38271547/when-should-we-call-multiprocessing-pool-join)

Reply:
Interesting!
I should investigate this and observe how performance evolves, will open an issue!